### PR TITLE
BaseRuleTest: assert evidence

### PR DESCRIPTION
**Pull request by idan-aviv**

### CODE REVIEW

File: tests/knowledge/rules/base_rule_test.py [hunk 1]
```diff
@@ -216,6 +217,9 @@ def _assert_rule(self, rule_result: RuleResponse, should_alert: bool,
                         f'The test {self.rule_under_test.get_id()} took {rule_runtime_seconds} seconds to run')
         test_function(self, rule_result)
 
+    def _assert_evidence(self, evidence_str: str):
+        self.assertRegexpMatches(evidence_str, r'^[A-Za-z0-9-_#?:\'".,\s`~/\\()\[\]*{}\$]+$', '')
```
Comment (Tomerva) [inline tests/knowledge/rules/base_rule_test.py:221]: what does the regex search for?

File: tests/knowledge/rules/base_rule_test.py [hunk 2]
```diff
@@ -208,6 +208,7 @@ def _assert_rule(self, rule_result: RuleResponse, should_alert: bool,
         if should_alert:
             self.assertEqual(RuleResultType.FAILED, rule_result.status)
             self.assertEqual(number_of_issue_items, len(rule_result.issues), rule_result.issues)
+            self._assert_evidence(rule_result.issues[0].evidence)
```
Comment (Viperain) [inline tests/knowledge/rules/base_rule_test.py:211]: go over all rule_result.issues not only the first one

**Review by Viperain — CHANGES_REQUESTED** [review]

File: tests/knowledge/rules/base_rule_test.py (hunk 1 above)
Comment (idan-aviv) [reply tests/knowledge/rules/base_rule_test.py:221]: @Tomerva invalid char in evidence string

File: tests/knowledge/rules/base_rule_test.py (hunk 2 above)
Comment (idan-aviv) [reply tests/knowledge/rules/base_rule_test.py:211]: NP but the evidence is the same! (besides friendly names)

**Review by Tomerva — APPROVED** [review]

**Review by Viperain — APPROVED** [review]

**Pull request merged**